src/server/game/World/World.cpp

```diff
@@ -3293,6 +3289,14 @@ CharacterInfo const* World::GetCharacterInfo(ObjectGuid const& guid) const
     return nullptr;
 }
 
+ObjectGuid World::GetCharacterGuidByName(std::string const& name) const
```

Comment:
Why did you choose World as the holder of this container?
IIRC everything is stored in ObjectMgr/ObjectAccessor